Sources/Target/FreeBSD/Process.cpp

```diff
@@ -201,8 +201,7 @@ ErrorCode Process::wait() {
 
   switch (_currentThread->_stopInfo.event) {
   case StopInfo::kEventNone:
-    switch (_currentThread->_stopInfo.reason) {
-    case StopInfo::kReasonNone:
+    if (_currentThread->_stopInfo.reason == StopInfo::kReasonNone) {
```

Comment:
This code makes no sense (the previous version). I'm not sure why we're checking `_stopInfo.reason`, so I think you can just get rid of the `if` block here. Look at the linux version of `Process:wait` and what we do when the stop event is `kEventNone`.